onnxmltools/utils/metadata_props.py

```diff
@@ -0,0 +1,67 @@
+from ..proto import onnx, onnx_proto
+from distutils.version import StrictVersion
+
+
+def _get_case_insensitive(iterable, key):
+    return next((x for x in iterable if x.lower() == key.lower()), None)
+
+
+def _validate_metadata(metadata_props):
+    valid_metadata_props = {
+        'Image.BitmapPixelFormat': ['Gray8', 'Rgb8', 'Bgr8', 'Rgba8', 'Bgra8'],
+        'Image.ColorSpaceGamma': ['Linear', 'SRGB'],
+        'Image.NominalPixelRange': ['NominalRange_0_255', 'Normalized_0_1', 'Normalized_1_1', 'NominalRange_16_235'],
+    }
+    image_metadata_props = {k: v for k, v in metadata_props.items() if _get_case_insensitive(valid_metadata_props, k)}
+    for key, value in image_metadata_props.items():
+        key = _get_case_insensitive(valid_metadata_props, key)
+        if not key:
+            print('Warning: key {} is defined multiple times'.format(key))
```

Comment:
Better to use `warning` module in python to create warning message. I am also not sure if it's just a warning or an exception.